tin/apps/assignments/views.py

```diff
@@ -463,7 +463,8 @@ def student_submissions_view(request, assignment_id, student_id):
 @login_required
 def submit_view(request, assignment_id):
     assignment = get_object_or_404(
-        Assignment.objects.filter_visible(request.user), id=assignment_id
+        Assignment.objects.filter_visible(request.user).filter_permission(request.user, "w"),
+        id=assignment_id,
```

Comment:
I think for queries like this we should have a `.filter_submittable` to make it easier to read.

You can also completely replace `filter_permission` with `filter_visible` (which already exists) and `filter_submittable`.

Reply:
I think it's a good idea to leave `filter_permissions`, just in case more flexibility is needed in the future. I just added aliases to it in `filter_submittable` instead.